service/service.go

```diff
@@ -269,3 +271,51 @@ func buildK8sRestConfig(config Config) (*rest.Config, error) {
 
 	return restConfig, nil
 }
+
+// NewClientCredentialsConfig returns a `ClientCredentialsConfig` configured taking values from Environment,
+// but operator parameters have precedence over environment variables.
+// These credentials will be used when talking to both Control Plane clusters and Tenant clusters.
+func NewClientCredentialsConfig(config Config) (auth.ClientCredentialsConfig, error) {
+	settings, err := auth.GetSettingsFromEnvironment()
+	if err != nil {
+		return auth.ClientCredentialsConfig{}, microerror.Mask(err)
+	}
+	if config.Viper.GetString(config.Flag.Service.Azure.ClientID) != "" {
+		settings.Values[auth.ClientID] = config.Viper.GetString(config.Flag.Service.Azure.ClientID)
+	}
+	if config.Viper.GetString(config.Flag.Service.Azure.ClientSecret) != "" {
+		settings.Values[auth.ClientSecret] = config.Viper.GetString(config.Flag.Service.Azure.ClientSecret)
+	}
+	if config.Viper.GetString(config.Flag.Service.Azure.TenantID) != "" {
+		settings.Values[auth.TenantID] = config.Viper.GetString(config.Flag.Service.Azure.TenantID)
+	}
+	if config.Viper.GetString(config.Flag.Service.Azure.SubscriptionID) != "" {
+		settings.Values[auth.SubscriptionID] = config.Viper.GetString(config.Flag.Service.Azure.SubscriptionID)
+	}
+
+	return settings.GetClientCredentials()
+}
+
+// NewCPAzureClientSet return an Azure client set configured for the control plane cluster.
+// If no control plane cluster tenant information has been passed to the operator, auth Azure Tenant will be used as control plane Azure Tenant.
+func NewCPAzureClientSet(config Config, gsClientCredentialsConfig auth.ClientCredentialsConfig) (*client.AzureClientSet, error) {
```

Comment:
A couple of new CLI parameters have been added to the operator so that we can pass the tenant id and subscription id for the control plane cluster.

If these new parameters are not passed to the operator, we use customer's single tenant service principal to create the Azure client set to manage the control plane. Same as we currently do.

If these new parameters are passed to the operator, we have tenants ids for both GiantSwarm and the customer. The code will create an Azure client set using GiantSwarm's Multi Tenant Service principal to manage the control plane.